### PR TITLE
Added checking of the Name and Value fields in the Expense model

### PR DESCRIPTION
**Pull request by KarolMaliglowka**

### CODE REVIEW

File: src/Contador.Core/Core/Models/Expense.cs [hunk 1]
```diff
@@ -78,5 +78,29 @@ public override int GetHashCode()
 		{
 			return base.GetHashCode();
 		}
+
+		public void SetName(string name)
```
Comment (JakubOciepa) [inline src/Contador.Core/Core/Models/Expense.cs:82]: Please make this function private.

File: src/Contador.Core/Core/Models/Expense.cs [hunk 2]
```diff
@@ -78,5 +78,29 @@ public override int GetHashCode()
...
+			}
+
+			if (name.Length < 3)
+			{
+				throw new ArgumentException("Name requires minimum 3 characters.");
+			}
+
+			Name = name;
+		}
+		public void SetValue(decimal value)
```
Comment (JakubOciepa) [inline src/Contador.Core/Core/Models/Expense.cs:96]: Please make this function private.

File: src/Contador.Core/Core/Models/Expense.cs [hunk 3]
```diff
@@ -78,5 +78,29 @@ public override int GetHashCode()
...
+				throw new ArgumentNullException(nameof(name), "Expense name cannot be empty.");
+			}
+
+			if (name.Length < 3)
+			{
+				throw new ArgumentException("Name requires minimum 3 characters.");
+			}
+
+			Name = name;
+		}
```
Comment (JakubOciepa) [inline src/Contador.Core/Core/Models/Expense.cs:95]: Please add empty line between methods.

File: src/Contador.Core/Core/Models/Expense.cs [hunk 4]
```diff
@@ -78,5 +78,29 @@ public override int GetHashCode()
...
+		public void SetValue(decimal value)
+		{
+			if (value < 0)
```
Comment (JakubOciepa) [inline src/Contador.Core/Core/Models/Expense.cs:98]: Here should be `if (value <= 0)` if this must be greater than 0.

**Review by JakubOciepa — COMMENTED** [review]
Thanks for your commitment ❤

File: src/Contador.Core/Core/Models/Expense.cs (hunk 1 above)
Comment (KarolMaliglowka) [reply src/Contador.Core/Core/Models/Expense.cs:82]: Changed access

File: src/Contador.Core/Core/Models/Expense.cs (hunk 2 above)
Comment (KarolMaliglowka) [reply src/Contador.Core/Core/Models/Expense.cs:96]: Changed access

File: src/Contador.Core/Core/Models/Expense.cs (hunk 3 above)
Comment (KarolMaliglowka) [reply src/Contador.Core/Core/Models/Expense.cs:95]: added

File: src/Contador.Core/Core/Models/Expense.cs (hunk 4 above)
Comment (KarolMaliglowka) [reply src/Contador.Core/Core/Models/Expense.cs:98]: Added

**Review by JakubOciepa — APPROVED** [review]
LGTM. Thanks again for the contribution.

**Pull request merged**